core/src/banking_stage/transaction_scheduler/transaction_state_container.rs

```diff
@@ -39,14 +40,16 @@ use {
 /// a new transaction, the lowest priority transaction will be dropped.
 pub(crate) struct TransactionStateContainer {
     priority_queue: MinMaxHeap<TransactionPriorityId>,
-    id_to_transaction_state: HashMap<TransactionId, TransactionState>,
+    id_to_transaction_state: Slab<TransactionState>,
 }
 
 impl TransactionStateContainer {
     pub(crate) fn with_capacity(capacity: usize) -> Self {
+        const EXTRA_CAPACITY: usize = 64;
+
         Self {
             priority_queue: MinMaxHeap::with_capacity(capacity),
-            id_to_transaction_state: HashMap::with_capacity(capacity),
+            id_to_transaction_state: Slab::with_capacity(capacity + EXTRA_CAPACITY),
```

Comment:
some small amount of extra capacity here because when we get a free index to insert (and then insert into queue/pop) we need somewhere to put it

Reply:
would be great to add this as a comment in the code